Src/AutoFixture/FreezeOnMatchCustomization.cs

```diff
@@ -75,6 +80,40 @@ public FreezeOnMatchCustomization(Type targetType)
         public IRequestSpecification Matcher { get; }
 
         /// <summary>
+        /// Used to create requests for resolving specimens before freezing.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// By default, the factory returns TargetType, but it can be tweaked 
+        /// to return any kind of request.
+        /// </para>
+        /// <para>
+        /// A typical use case is to create a request which includes significant 
+        /// information about a specimen. For instance, <see cref="ParameterInfo"/> 
+        /// which is widely used in Glue Libraries can contain custom attributes
+        /// such as <see cref="StringLengthAttribute"/> or <see cref="RegularExpressionAttribute" />.
+        /// These attributes are essential part of the request since they influence 
+        /// the final specimen.
+        /// </para>
+        /// </remarks>
+        public Func<object> RequestFactory
```

Comment:
It appeared to me that this factory is a method which _emphasizes_ its goal to crate requests. The difference with a property is minimal and I don't have a strong opinion that it must be a delegate.
No other reasons. Would you prefer to convert it into `Request` property?